src/node_buffer.cc

```diff
@@ -541,6 +662,112 @@ void WriteDoubleBE(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+template <typename T, enum Endianness endianness, T (*strtoT)(const char *, char **, int)>
+uint32_t WriteInt64Generic(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  ARGS_THIS(args[0].As<Object>())
+
+  T val;
+  if (args[1]->IsNumber()) {
+    val = args[1]->IntegerValue();
+  } else if (args[1]->IsString()) {
+    // Have to do this because strtoll/strtoull doesn't set errno to 0 on success
+    errno = 0;
+    v8::String::Utf8Value str(args[1]);
+    val = strtoT(*str, NULL, 0);
```

Comment:
Why don't you pass in an endptr?  You don't have to clobber errno that way.

Also, would it make sense to specify the numeric base?